File: market/test/java/ru/yandex/market/wms/timetracker/utils/SecurityUtil.java
package ru.yandex.market.wms.timetracker.utils;

import java.util.List;
import java.util.Objects;

import org.springframework.security.core.context.SecurityContext;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;

import ru.yandex.market.wms.auth.core.model.InforAuthentication;

public class SecurityUtil {

    private static final String ATTR = "SPRING_SECURITY_CONTEXT";

    private SecurityUtil() {

    }

    public static void setSecurityContextAttribute(MockHttpServletRequestBuilder requestBuilder, String username) {
        requestBuilder.sessionAttr(
                ATTR,
                createSecurityContext(Objects.requireNonNullElse(username, "defaultTestUser"))
        );
    }

    private static SecurityContext createSecurityContext(String username) {
        var tokenContent = username + "-token";
        var token = new InforAuthentication(username, tokenContent, List.of());
        token.setAuthenticated(true);

        var context = SecurityContextHolder.getContext();
        context.setAuthentication(token);
        return context;
    }
}

File: market/test/kotlin/ru/yandex/market/wms/taskrouter/util/SecurityUtil.kt
package ru.yandex.market.wms.taskrouter.util

import org.springframework.security.core.context.SecurityContext
import org.springframework.security.test.context.TestSecurityContextHolder
import org.springframework.test.web.servlet.MockHttpServletRequestDsl
import ru.yandex.market.wms.auth.core.model.InforAuthentication

private const val ATTR = "SPRING_SECURITY_CONTEXT"

fun MockHttpServletRequestDsl.setSecurityContextAttribute(username: String = "defaultTestUser") {
    sessionAttr(ATTR, createSecurityContext(username))
}

private fun createSecurityContext(username: String): SecurityContext {
    val tokenContent = "$username-token"
    val token = InforAuthentication(username, tokenContent, emptyList()).apply { isAuthenticated = true }
    return TestSecurityContextHolder.getContext().apply { authentication = token }
}
